yt_dlp/extractor/nebula.py

```diff
@@ -120,50 +54,25 @@ def _retrieve_nebula_auth(self):
 
         return response['key']
 
-    def _retrieve_zype_api_key(self, page_url, display_id):
-        """
-        Retrieves the Zype API key
-        """
-
-        # Find the js that has the API key from the webpage and download it
-        webpage = self._download_webpage(page_url, video_id=display_id)
-        main_script_relpath = self._search_regex(
-            r'<script[^>]*src="(?P<script_relpath>[^"]*main.[0-9a-f]*.chunk.js)"[^>]*>', webpage,
-            group='script_relpath', name='script relative path', fatal=True)
-        main_script_abspath = urljoin(page_url, main_script_relpath)
-        main_script = self._download_webpage(main_script_abspath, video_id=display_id,
-                                             note='Retrieving Zype API key')
-
-        api_key = self._search_regex(
-            r'REACT_APP_ZYPE_API_KEY\s*:\s*"(?P<api_key>[\w-]*)"', main_script,
-            group='api_key', name='API key', fatal=True)
-
-        return api_key
-
-    def _call_zype_api(self, path, params, video_id, api_key, note):
-        """
-        A helper for making calls to the Zype API.
-        """
-        query = {'api_key': api_key, 'per_page': 1}
-        query.update(params)
-        return self._download_json('https://api.zype.com' + path, video_id, query=query, note=note)
-
-    def _call_nebula_api(self, path, video_id, access_token, note):
-        """
-        A helper for making calls to the Nebula API.
-        """
-        return self._download_json('https://api.watchnebula.com/api/v1' + path, video_id, headers={
-            'Authorization': 'Token {access_token}'.format(access_token=access_token)
-        }, note=note)
-
-    def _fetch_zype_access_token(self, video_id):
+    def _call_nebula_api(self, url, video_id=None, method='GET', auth_type='api', note=''):
+        assert method in ('GET', 'POST',)
+        assert auth_type in ('api', 'bearer',)
+        authorization = 'Token {api_token}'.format(api_token=self._nebula_api_token) \
+            if auth_type == 'api' \
+            else 'Bearer {bearer_token}'.format(bearer_token=self._nebula_bearer_token)
+        url_or_request = url \
+            if method == 'GET' \
+            else compat_urllib_request.Request(url, method='POST', data={})
+        return self._download_json(url_or_request, video_id, headers={'Authorization': authorization}, note=note)
```

Comment:
```suggestion
        authorization = f'Token {self._nebula_api_token}' if auth_type == 'api' else f'Bearer {self._nebula_bearer_token}'
        return self._download_json(
            url, video_id, note=note, headers={'Authorization': authorization},
            data=b'' if method == 'POST' else None)
```

Reply:
Not sure I understand the final bit here. Are you saying that it will automatically be a POST request if I specify `data`, and therefore I don't need to construct my own `Request` object anymore?

Reply:
yes. It is the behavior of urllib to use POST if data is given and GET otherwise

Reply:
Awesome, that makes this indeed a lot simpler!